src/mongo_client.ts

```diff
@@ -364,6 +364,45 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         return true;
       }
     };
+    this.checkForNonGenuineHosts();
+  }
+
+  DOCUMENT_DB_CHECK = /(\.docdb\.amazonaws\.com$)|(\.docdb-elastic\.amazonaws\.com$)/;
+  COSMOS_DB_CHECK = /\.cosmos\.azure\.com$/;
+
+  static DOCUMENT_DB_MSG =
+    'You appear to be connected to a DocumentDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/documentdb';
+  static COSMOS_DB_MSG =
+    'You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb';
```

Comment:
I'd prefer these to all be `const` outside of the `MongoClient` itself, since they aren't really properties of the class or its prototype. And now that I think about this more, I think this functionality is probably all better served in the utils module. So could we move the constants there as well as the `isHostMatch()` function?

Reply:
That makes sense!